src/app/icd/client/DefaultICDClientStorage.cpp

```diff
@@ -39,6 +39,9 @@ static_assert(kMaxFabricListTlvLength <= std::numeric_limits<uint16_t>::max(), "
 
 namespace chip {
 namespace app {
+
+#define APPDATA_LENGTH 6
```

Comment:
I think this should a const expr to be inline with c++ standards.
and the constant should be in the generic ClientStorage imo.